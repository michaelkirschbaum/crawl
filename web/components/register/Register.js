import React, { Component } from 'react'
import './Register.css'

class Register extends Component {
  constructor(props) {
    super(props)
    this.state = {
      first: '',
      last: '',
      email: '',
      password: ''
    }
    this.handleSubmit = this.handleSubmit.bind(this)
    this.onChange = this.onChange.bind(this)
  }

  handleSubmit(event) {
    event.preventDefault()

    const { first, last, email, password } = this.state
    fetch(`http://localhost:8080/users/add?first=${first}&last=${last}&email=${email}&password=${password}`, { method: "POST" })
      .then(res => this.props.history.push('/'))
      .catch(err => console.log(err))
  }

  onChange(event) {
    this.setState({[event.target.name]: event.target.value})
  }

  render() {
    return (
      <div className="Register">
        <h1>Register</h1>
        <form onSubmit={this.handleSubmit}>
          <input type="text" name="first" onChange={this.onChange} placeholder="First name"/><br />
          <input type="text" name="last" onChange={this.onChange} placeholder="Last name"/><br />
          <input type="text" name="email" onChange={this.onChange} placeholder="Email"/><br />
          <input type="password" name="password" onChange={this.onChange} placeholder="Password"/><br />
          <input type="submit" value="Register"/>
        </form>
      </div>
    )
  }
}

export default Register
