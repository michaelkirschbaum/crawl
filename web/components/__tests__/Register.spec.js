import React from 'react'
import { Register } from '../../components/register/Register'
import { shallow } from 'enzyme'

describe('Register', () => {
  it('should render correctly', () => {
    const component = shallow(<Register />)
    expect(component).toMatchSnapshot()
  })
})
