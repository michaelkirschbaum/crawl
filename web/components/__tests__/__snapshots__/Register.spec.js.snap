// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Register
        </h1>,
        <form
          onSubmit={[Function]}
        >
          <input
            name="first"
            onChange={[Function]}
            placeholder="First name"
            type="text"
          />
          <br />
          <input
            name="last"
            onChange={[Function]}
            placeholder="Last name"
            type="text"
          />
          <br />
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <br />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="text"
          />
          <br />
          <input
            type="submit"
            value="Register"
          />
        </form>,
      ],
      "className": "Register",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Register",
        },
        "ref": null,
        "rendered": "Register",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="first"
              onChange={[Function]}
              placeholder="First name"
              type="text"
            />,
            <br />,
            <input
              name="last"
              onChange={[Function]}
              placeholder="Last name"
              type="text"
            />,
            <br />,
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />,
            <br />,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="text"
            />,
            <br />,
            <input
              type="submit"
              value="Register"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "first",
              "onChange": [Function],
              "placeholder": "First name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "last",
              "onChange": [Function],
              "placeholder": "Last name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Register",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Register
          </h1>,
          <form
            onSubmit={[Function]}
          >
            <input
              name="first"
              onChange={[Function]}
              placeholder="First name"
              type="text"
            />
            <br />
            <input
              name="last"
              onChange={[Function]}
              placeholder="Last name"
              type="text"
            />
            <br />
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
            <br />
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="text"
            />
            <br />
            <input
              type="submit"
              value="Register"
            />
          </form>,
        ],
        "className": "Register",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Register",
          },
          "ref": null,
          "rendered": "Register",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="first"
                onChange={[Function]}
                placeholder="First name"
                type="text"
              />,
              <br />,
              <input
                name="last"
                onChange={[Function]}
                placeholder="Last name"
                type="text"
              />,
              <br />,
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
              />,
              <br />,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="text"
              />,
              <br />,
              <input
                type="submit"
                value="Register"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "first",
                "onChange": [Function],
                "placeholder": "First name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "last",
                "onChange": [Function],
                "placeholder": "Last name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Register",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
